Fable.Remoting.AspNetCore/Middleware.fs

```diff
@@ -0,0 +1,151 @@
+                return Some ctx 
+            }
+
+    let setContentType (contentType: string) : HttpHandler = 
+        fun (next : HttpFunc) (ctx : HttpContext) -> 
+            task {
+                ctx.Response.ContentType <- contentType
+                return Some ctx 
+            }
+
+    let success value : HttpHandler = 
+        setResponseBody value 
+        >=> setStatusCode 200 
+        >=> setContentType "application/json; charset=utf-8"
+
+    let failure error = 
+       setResponseBody error
+       >=> setStatusCode 500 
+       >=> setContentType "application/json; charset=utf-8"
+    
+    /// Used to forward of the Http context
+    let halt : HttpHandler = 
+      fun (next : HttpFunc) (ctx : HttpContext) ->
+        task { return None }
+
+    let fail (ex: exn) (routeInfo: RouteInfo<HttpContext>) (options: RemotingOptions<HttpContext, 't>) : HttpHandler = 
+      fun (next : HttpFunc) (ctx : HttpContext) -> 
+        task {
+            match options.ErrorHandler with 
+            | None -> return! failure (Errors.unhandled routeInfo.methodName) next ctx 
+            | Some errorHandler -> 
+                match errorHandler ex routeInfo with 
+                | Ignore -> return! failure (Errors.ignored routeInfo.methodName) next ctx  
+                | Propagate error -> return! failure (Errors.propagated error) next ctx  
+        }
+
+    let runFunction func impl options args : HttpHandler = 
+      fun (next : HttpFunc) (ctx : HttpContext) -> 
+        task {
+            let! functionResult = Async.StartAsTask (Async.Catch (DynamicRecord.invokeAsync func impl args)) 
+            match functionResult with
+            | Choice.Choice1Of2 output -> return! success output next ctx 
+            | Choice.Choice2Of2 ex -> 
+              let routeInfo = { methodName = func.FunctionName; path = ctx.Request.Path.ToString(); httpContext = ctx }
+              return! fail ex routeInfo options next ctx 
+        }
+
+    let buildFromImplementation impl options = 
+      let dynamicFunctions = DynamicRecord.createRecordFuncInfo impl
+      let typeName = impl.GetType().Name   
+      fun (next : HttpFunc) (ctx : HttpContext) -> task {
+        let foundFunction = 
+          dynamicFunctions 
+          |> Map.tryFindKey (fun funcName _ -> ctx.Request.Path.Value = options.RouteBuilder typeName funcName) 
+        match foundFunction with 
+        | None -> return! halt next ctx   
+        | Some funcName -> 
+            let func = Map.find funcName dynamicFunctions
+            match ctx.Request.Method.ToUpper(), func.Type with  
+            | "GET", NoArguments _ ->  
+                return! runFunction func impl options [|  |] next ctx  
+            | "GET", SingleArgument(input, _) when input = typeof<unit> ->
+                return! runFunction func impl options [|  |] next ctx    
+            | "GET", SingleArgument(input, _) when input = typeof<unit> -> 
```

Comment:
Is this a duplication of lines 106-107? I guess this was planned to be "POST" ?

Reply:
You are correct, nice catch!